File: LAB2/app/src/main/java/com/example/tipper/RecipeActivity.java
package com.example.tipper;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageButton;

public class RecipeActivity extends AppCompatActivity {
    Button btnBack;
    ImageButton firstRecipe, secondRecipe;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_recipe);

        btnBack = (Button) findViewById(R.id.btnBack);
        btnBack.setOnClickListener(new View.OnClickListener(){
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(RecipeActivity.this, MainActivity.class);
                startActivity(intent);
            }
        });

        firstRecipe = (ImageButton) findViewById(R.id.firstRecipe);
        firstRecipe.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(RecipeActivity.this, FirstRecipeActivity.class);
                startActivity(intent);
            }
        });

        secondRecipe = (ImageButton) findViewById(R.id.secondRecipe);
        secondRecipe.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(RecipeActivity.this, SecondRecipeActivity.class);
                startActivity(intent);
            }
        });
    }


}

File: LAB4/app/src/main/java/com/example/myapplicationpamo/RecipeActivity.kt
package com.example.myapplicationpamo

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ImageButton

class RecipeActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_recipe)

        val firstRecipe = findViewById<ImageButton>(R.id.firstRecipe)
        firstRecipe.setOnClickListener { displayFirstrecipe() }

        val secondRecipe = findViewById<ImageButton>(R.id.secondRecipe)
        secondRecipe.setOnClickListener { displaySecondRecipe() }
    }

    private fun displayFirstrecipe() {
        val intent = Intent(this, FirstRecipeActivity::class.java)
        startActivity(intent)
    }
    private fun displaySecondRecipe() {
        val intent = Intent(this, SecondRecipeActivity::class.java)
        startActivity(intent)
    }
}